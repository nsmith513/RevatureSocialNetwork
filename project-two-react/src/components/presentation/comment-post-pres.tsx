import React, { useEffect, useState } from 'react';
import axios  from 'axios';
import '../styles/comment-post-pres.css';

interface IProps {
    
}

export const CommentPost: React.FC<IProps> = (props:IProps) => {

    const [comment, setComment] = useState('');

    const onContentChange = (evt:any) => {
        setComment(evt.target.value);
    }

    const handleSubmit = (evt:any) => {
        evt.preventDefault();

        const commentObj = {
            content: comment,
            user: 1,
            post: 2
        
        }
        axios.post(process.env.REACT_APP_API_URL + 'ProjectTwoSTS/api/post/comment',commentObj)
        .then(res => console.log(res))
        .catch(err => console.log(err)
        );
    }

    return (
        <form onSubmit={handleSubmit}>
            <div className="d-flex justify-content-center">
            <div className="card register-card">
                <div className="card-header">
                    <h3>Comment</h3>
                </div>
                <div className="card-body">
                    <div className="mb-3">
                        <textarea  onChange={onContentChange} value={comment} rows={5} className="form-control" name="description" placeholder="Write a post..."></textarea>
                    </div>
                    
                    <div className="form-group">
                        
                    </div>

                    <button type="submit" className="btn btn-primary" id="input-creds">Submit</button>
                    </div>
                </div>
            </div>
        </form>
    );
}