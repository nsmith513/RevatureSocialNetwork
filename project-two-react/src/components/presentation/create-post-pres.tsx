import React, { FormEvent, RefObject, useEffect, useState } from 'react';
import axios  from 'axios';
import '../styles/create-post-pres.css';

interface IProps {
    fileInput: any;
    handleSubmit: ((event: FormEvent<HTMLFormElement>) => void) | undefined;
    setPostValue(value: string): void;
    postContent: string;

}

export const CreatePost: React.FC<IProps> = (props:IProps) => {



    return (
        <form onSubmit={props.handleSubmit}>
            <div className="container">
                <div className="card create-post">

                    <div id="inputsContainer1" className="col-md-8 col-md-offset-2">

                        <h1 className="createPostHeader">Create a post</h1>

                        <div id="create-post-card">
                            <div className="form-group">
                                <textarea value={props.postContent} onChange={(evt) => props.setPostValue(evt.target.value)} rows={5} className="form-control" name="description" placeholder="Write a post..."></textarea>
                            </div>

                            <div>
                            <label htmlFor="formFileSm" className="form-label">Add an image</label>
                            <input ref={props.fileInput} className="form-control postInputForm form-control-sm" id="formFileSm" type="file" />
                            </div>
                            
                            <br/>

                            <div className="form-group">
                            <button className="btn btn-outline-danger cancleCreateButton">
                                    Cancel
                                </button>
                                <button type="submit" className="btn btn-outline-success createPostButton">
                                    Create
                                </button>

                            </div>

                        </div>
                    </div>

                </div>
            </div>
        </form>
    );
}