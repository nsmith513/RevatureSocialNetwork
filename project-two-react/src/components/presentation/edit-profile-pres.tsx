import React from 'react';
import IUser from '../../models/IUser';
import '../styles/edit-profile-pres.css';

interface IUserModel {
        id:number,
        uname:string,
        pwd:string,
        email:string,
        pfp:string
}

interface IProps {
    handleSubmit(event: React.FormEvent<HTMLFormElement>): void;
    setUsername: Function;
    setPassword: Function;
    setEmail: Function;
    username: string;
    password: string;
    email: string,
    userData: IUserModel;
    fileInput:any;
}

export const EditProfilePres: React.FC<IProps> = (props:IProps) => {
    return(
        <>
         <form onSubmit={props.handleSubmit}>
            <div className="card edit-profile-card">
                    <div className="card-body userProfile">

                    <h1>Update Profile</h1>
                        
                            <img className='avatar updateProfileAvatarImage  edit-custom-img' src={props.userData.pfp}/> 

                            <div className="">
                                <label htmlFor="formFileSm" className="form-label">Change your profile picture</label>
                                <input ref={props.fileInput} className="updateProfileImage form-control form-control-sm" id="formFileSm" type="file" />
                            </div>

                            <div className="mb-3">
                                <label htmlFor="exampleFormControlInput1" className="form-label">Change Username</label>
                                <input value={props.username} onChange={(evt) => props.setUsername(evt.target.value)} type="text"  name="username" className="form-control" id="exampleFormControlInput1" placeholder={props.userData.uname} />
                            </div>

                            <div className="mb-3">
                                <label htmlFor="exampleFormControlInput2" className="form-label">Change Email</label>
                                <input value={props.email} onChange={(evt) => props.setEmail(evt.target.value)} type="email"  name="email" className="form-control" id="exampleFormControlInput2" placeholder={props.userData.email} />
                            </div>

                            <div className="mb-3">
                                <label htmlFor="exampleFormControlInput2" className="form-label">Change Password</label>
                                <input value={props.password} onChange={(evt) => props.setPassword(evt.target.value)} type="password"  name="password" className="form-control" id="exampleFormControlInput3" placeholder='password' />
                            </div>

                            <div className="mb-3 profileSubmitButtonContainer">
                                <button type="submit" className="btn btn-outline-light mb-3 updateProfileSubmitButton">Update Info</button>
                            </div>
                                    
                            <br/>
                        
                    </div>
                </div>
            </form>
        </>
    )
}