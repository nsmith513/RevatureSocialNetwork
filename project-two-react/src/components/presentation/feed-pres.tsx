import React from 'react';

export const FeedPres: React.FC = (props) =>{
    return <>{props.children}</>;
}
