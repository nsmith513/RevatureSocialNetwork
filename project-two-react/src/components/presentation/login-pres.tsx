import React, { useState } from 'react';
import '../styles/login-pres.css';
import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';

interface IProps {
    handleSubmit(event: React.FormEvent<HTMLFormElement>): void;
    setUsername: Function;
    setPassword: Function;
    username: string;
    password: string;
}

export const LoginPres: React.FC<IProps> = (props:IProps) =>{

    return (
        <div className="login">
            <div className="d-flex justify-content-center">
                <div className="card login-card">
                    <div className="card-header">
                        <h3 className="loginCardHeader">User Login</h3>
                    </div>
                    <div className="card-body">

                        <form onSubmit={props.handleSubmit}>
                            <div className="mb-3">

                                <label htmlFor="inputUsername1" className="form-label ">Username</label>
                                <input placeholder="Username" onChange={(evt) => props.setUsername(evt.target.value)} type="text" className="form-control form-control-lg" id="input-username" name="username" value={props.username} />
                            </div>
                            <div className="mb-3">
                                <label htmlFor="inputPassword1" className="form-label">Password</label>
                                <input placeholder="Password" onChange={(evt) => props.setPassword(evt.target.value)} type="password" className="form-control form-control-lg" id="input-password" name="password" value={props.password} />
                            </div>
                            <div className="row">
                                <div className="col">

                                    <button type="submit" className="btn btn-lg btn-outline-light loginSubmitButton" id="input-creds">Submit</button>

                                </div>
                                <div className="col text-right">
                                    <Link to="/register"><button className="btn btn-lg btn-outline-warning registerButton goToRegisterButton">Register</button></Link>
                                </div>
                            </div>
                        </form>

                    </div>
                </div>
            </div>
        </div>
    );

    

}
