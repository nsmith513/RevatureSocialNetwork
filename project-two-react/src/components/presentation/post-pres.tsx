import React from 'react';
import IPost from '../../models/IPost';
import '../styles/post-pres.css';

interface IProps {
    likeButtonClick(event: any): void;
    post: IPost;
    onOff: Boolean;
    likeCount: number;
}

interface IImages{
    id: number,
    link: string
}

export const PostPres: React.FC<IProps> = (props: IProps) => {
    return props.post.user ?
        <>
            <div className="card post-custom-card ">
                <div className="card-body">
                    <div className="clearfix">
                        <img className='avatar pull-left custom-img' src={props.post.user.pfp} />
                        <h5 className="card-title">{props.post.user.uname}</h5>
                    </div>
                    <p className="card-text">{props.post.content}</p>

                    <div>{props.post.images.map((image) => <img className="img-fluid img-thumbnail" key={image.id} src={image.link}/>)}</div>
                    
                    <hr />
                    <button onClick={props.likeButtonClick}
                        style={props.onOff ? { borderColor: 'blue' } : { borderColor: 'grey' }}
                        className="btn btn-link">
                        Like {props.likeCount > 0 ? props.likeCount : ""}
                    </button>
                    {/* <a href="#" className="card-link">Comment</a> */}
                </div>
            </div>
        </> :
        <>
            <h2>No posts</h2>
        </>
}
