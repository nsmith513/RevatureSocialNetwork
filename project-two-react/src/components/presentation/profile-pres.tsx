import React from 'react';
import '../styles/profile-pres.css';

interface IProps {
    user : {
        uname:string,
        email:string,
        pfp:string
    }
}

export const ProfilePres : React.FC<IProps> = (props:IProps) => {
    return props.user ?
        <>
            <div className="container container-profile">
                <h3 className="profileHeader">Profile</h3>
                <div className="row">
                    <div className="col-sm leftProfileColumn">
                        <img className='avatar' src={props.user.pfp} alt="PFP" height="100rem"/>
                        < hr className="horizontalRule" />

                    </div>
                    <div className="col-sm rightProfileColumn">
                    <span className="profileUserName" >Username: <br/><span className="userProfileInfo">{props.user.uname}</span></span>
                        <br/>
                        Email: <br/><span className="userProfileInfo">
                        {props.user.email} </span>
                    </div>
                </div>
            </div>
        </> :
        <>
            <p>No user</p>
        </>;
}
