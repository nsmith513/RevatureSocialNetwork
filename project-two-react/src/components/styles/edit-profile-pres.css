.edit-profile-card {
  width: 30rem;
  background-color: var(--col-e);
  margin: 0 auto;
  border-radius: 2em;
}

.edit-custom-img {
  width: 10rem;
  height: 10rem;
}

.userProfile {
  display: inline-grid;
}

.userProfile > h1 {
  margin-left: auto;
  margin-right: auto;
  font-weight: bold;
  -webkit-text-stroke-width: 0.6px;
  -webkit-text-stroke-color: #477387;
}
.updateProfileAvatarImage {
  margin-left: auto;
}
.updateProfileImage {
  width: 200px;
  height: 36px;
}

.profileSubmitButtonContainer {
}
.updateProfileSubmitButton {
  margin-left: auto;
  width: 100%;
}
