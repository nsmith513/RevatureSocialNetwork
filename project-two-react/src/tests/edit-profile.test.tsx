import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {EditProfilePres} from '../components/presentation/edit-profile-pres';

Enzyme.configure({ adapter: new Adapter() })


const fileInput = null;
const handleSubmit = (evt: any) => {}
const setUsername = () => {}
const setPassword = () => {}
const setEmail = () => {}
const username = ''
const password = ''
const email =''
const userData = {id:0,uname:'',email:'',pfp:'',pwd:''};

describe('<EditProfilePres />', () => {
    const container = shallow(<EditProfilePres
        handleSubmit={handleSubmit}
        setUsername={setUsername}
        setPassword={setPassword}
        setEmail={setEmail}
        username={username}
        password={password}
        email={email}
        userData={userData}
        fileInput={fileInput}
       />);

    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });
});