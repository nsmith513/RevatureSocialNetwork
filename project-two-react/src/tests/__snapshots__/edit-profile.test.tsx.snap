// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditProfilePres /> should match the snapshot 1`] = `"<form><div class=\\"card edit-profile-card\\"><div class=\\"card-body userProfile\\"><h1>Update Profile</h1><img class=\\"avatar updateProfileAvatarImage  edit-custom-img\\" src=\\"\\"/><div class=\\"\\"><label for=\\"formFileSm\\" class=\\"form-label\\">Change your profile picture</label><input type=\\"file\\" class=\\"updateProfileImage form-control form-control-sm\\" id=\\"formFileSm\\"/></div><div class=\\"mb-3\\"><label for=\\"exampleFormControlInput1\\" class=\\"form-label\\">Change Username</label><input type=\\"text\\" value=\\"\\" name=\\"username\\" class=\\"form-control\\" id=\\"exampleFormControlInput1\\" placeholder=\\"\\"/></div><div class=\\"mb-3\\"><label for=\\"exampleFormControlInput2\\" class=\\"form-label\\">Change Email</label><input type=\\"email\\" value=\\"\\" name=\\"email\\" class=\\"form-control\\" id=\\"exampleFormControlInput2\\" placeholder=\\"\\"/></div><div class=\\"mb-3\\"><label for=\\"exampleFormControlInput2\\" class=\\"form-label\\">Change Password</label><input type=\\"password\\" value=\\"\\" name=\\"password\\" class=\\"form-control\\" id=\\"exampleFormControlInput3\\" placeholder=\\"password\\"/></div><div class=\\"mb-3 profileSubmitButtonContainer\\"><button type=\\"submit\\" class=\\"btn btn-outline-light mb-3 updateProfileSubmitButton\\">Update Info</button></div><br/></div></div></form>"`;
